### ISSUE TITLE
Parallel model implementation

### ISSUE THREAD

**Issue by eddie-c-davis** [MeteoSwiss-APN/dawn#753]
Complete the implementation of the Dawn parallel execution model.
* Ensure that existing passes (stage merger, multistage merger, etc.) are consistent with the model
* Document the model and post to the Wiki
* Merge the IR lowering code (SIR -> IIR)
* Communicate with collaborators (e.g., GT4Py team)

**Comment by cosunae** [comment]
@eddie-c-davis I think this was decomposed and cover by other tasks, can we close it?

**Comment by eddie-c-davis** [comment]
Yes, @twicki has created an epic for this in issue #836, so I believe this one can be closed.
